airbyte-integrations/bases/connector-acceptance-test/connector_acceptance_test/tests/test_core.py

```diff
@@ -488,11 +488,14 @@ def test_check(self, connector_config, inputs: ConnectionTestConfig, docker_runn
             assert len(con_messages) == 1, "Connection status message should be emitted exactly once"
             assert con_messages[0].connectionStatus.status == Status.FAILED
         elif inputs.status == ConnectionTestConfig.Status.Exception:
-            with pytest.raises(ContainerError) as err:
-                docker_runner.call_check(config=connector_config)
-
-            assert err.value.exit_status != 0, "Connector should exit with error code"
-            assert "Traceback" in err.value.stderr, "Connector should print exception"
+            output = docker_runner.call_check(config=connector_config, raise_container_error=False)
+            trace_messages = filter_output(output, Type.TRACE)
+            assert len(trace_messages) == 1, "A trace message should be emitted in case of unexpected errors"
+            trace = trace_messages[0].trace
+            assert isinstance(trace, AirbyteTraceMessage)
+            assert trace.error is not None
+            assert trace.error.message is not None
+            assert trace.error.internal_message is not None
```

Comment:
👍🏻 I like this test because it doesn't make any opinions about whether a `connection_status` message is output, only confirms that a trace message is output. This follows the behavior that I now use in the platform, where if there was a trace message, we declare that there was an error and don't take a connection_status message (if existing) into account. So connectors that currently output both won't fail the test. 

Are we concerned about connectors that aren't CDK ones that have the same behavior (e.g. most destinations - not sure if we regularly run CAT on destinations but we could) will suddenly start failing this?

Reply:
I guess this is only tested if someone provides a `status: "exception"` config to test though

Reply:
> I guess this is only tested if someone provides a status: "exception" config to test though

Yes indeed.
I'd love to find a way to create a global not configurable test for this. Because it's unlikely developers will fill in a `status: "exception"` as they'll rather use `status: "fail"` to make sure the expected errors are correctly handled.
Do you have a solution in mind to always make check fail in this test context? (Maybe preventing internet access to the container under test?)

Reply:
I believe check currently fails if we pass in a path to a config that doesn't exist. We could try to leverage that?

Reply:
Yep this could work. The problem is that `test_check` relies on the `inputs` fixture which is reading the config. Passing an unexisting config leads to a FileNotFoundError on CAT.
I'll try to create a separate test not using this fixture.

Reply:
Haha, our fixtures are too good to make fake errors 😄